src/hxparse/LexEngine.hx

```diff
@@ -331,6 +331,29 @@ class LexEngine {
 	}
 
 	static function parseInner( pattern : byte.ByteData, i : Int = 0, pDepth : Int = 0 ) : { pattern: Pattern, pos: Int } {
+		function readChar() {
+			var c = pattern.readByte(i++);
+			if ( StringTools.isEof(c) ) {
+				c = '\\'.code;
+			} else if (c == "x".code) {
+				c = Std.parseInt("0x" + pattern.readString(i, 2));
```

Comment:
Should this really be restricted to 2 characters? If we're talking Unicode you might need 4 here. Then again I don't really know how to determine that easily.